v2.2/architecture/life-of-a-distributed-transaction.md

```diff
@@ -0,0 +1,204 @@
+
+<!-- #### Transaction Record
+
+The first range that a transaction touches receives an additional KV operation to create a translation record (`TxnRecord`). This key-value pair is used to store the transaction's current state for the life of the transaction. It starts off in a `PENDING` state, and then progresses to either `COMMITTED` if the transaction succeeds or `ABORTED` if the transaction must be canceled.  -->
+
+## Leaseholder node
+
+The gateway node's `DistSender` tries to send its `BatchRequests` to the replica identified as the range's [leaseholder](replication-layer.html#leases), which is a single replica that serves all reads for a range, as well as coordinates all writes. Leaseholders play a crucial role in CockroachDB's architecture, so it's a good topic to make sure you're familiar with.
+
+### Request response
+
+Because the leaseholder replica can shift between nodes, all nodes must be able to return a request for any key.
+
+##### No Longer Leaseholder
+
+If a node:
+
+- *Was* the leaseholder, but no longer holds the lease
+- Holds a replica for the range
+
+Then the node denies the request, but includes the last known address for the leaseholder of that range.
+
+Upon receipt of this response, the `DistSender` will update the header of the `BatchRequest` with the new address, and then resend the `BatchRequest` to the newly identified leaseholder.
+
+##### No Longer Has/Never Had Range
+
+If a node doesn't have a replica for the requested range, it denies the request without providing any further information.
+
+In this case, the `DistSender` must look up the current leaseholder using the [cluster's `meta` ranges](distribution-layer.html#meta-range-kv-structure) range.
+
+##### Success
+
+Once the node that contains the leaseholder of the range receives the `BatchRequest`, it begins processing it, and progresses onto checking the timestamp cache.
+
+### Timestamp cache
+
+The timestamp cache tracks the highest timestamp (i.e., most recent) for any read operation that a given range has served.
+
+Each write operation in a `BatchRequest` checks its own timestamp versus the timestamp cache to ensure that the write operation has a higher timestamp; this guarantees that history is never rewritten and you can trust that reads always served the most recent data. If a write operation fails this check, it must be restarted at a timestamp higher than the timestamp cache's value.
+
+### Latch manager
+
+Operations in the `BatchRequest` are serialized through the leaseholder's latch manager.
+
+This works by giving each write operation a latch on a row. Any reads or writes that come in after the latch has been granted on the row must wait for the write to complete, at which point the latch is released and the subsequent operations can continue.
+
+### Evaluator
```

Comment:
I'm not familiar with this term in the storage layer, although there is a SQL evaluator, which I think you're referring to. `CHECK` constraints are validated at that level, which is up above the `TxnCoordSender`.

Reply:
You might also be referring to "batch evaluation", which is where `BatchRequests` determine their result under the "protection" of the latch manager. I'd suggest renaming this to `Batch Evaluation` and removing any reference to `CHECK` cosntraints.